hack/pre-release.sh

```diff
@@ -0,0 +1,109 @@
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+  
+  # Find the previous version from the olm.skipRange in cluster_service_version.yaml
+  last_version=$(grep "olm.skipRange" $cluster_service_version_yaml | 
+    grep -oP ">=\K([0-9]+\.[0-9]+\.[0-9]+)\s*<" | sed "s/\s*<//" | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  echo "WMCO version $version detected from the Makefile"
+
+  echo "Updating the Makefile to version $updated_version"
+  sed -i "s/$version/$updated_version/g" Makefile
+ 
+  echo "Updating the cluster service version yaml to $updated_version"
+  sed -i "s/$version/$updated_version/g" $cluster_service_version_yaml
+  
+  echo "Updating previous version to $version"
+  sed -i "s/$last_version/$version/g" $cluster_service_version_yaml
+
+  echo "Updating Dockerfile version to $updated_version"
+  sed -i "s/$version/$updated_version/g" bundle.Dockerfile 
+
+  echo "Updating annotation version to $updated_version"
+  sed -i "s/$version/$updated_version/g" bundle/metadata/annotations.yaml 
+
+  echo "Updating config-operator.package.yaml to $updated_version" 
+  sed -i "s/$version/$updated_version/g" bundle/windows-machine-config-operator.package.yaml
+}
+
+update_OCP_version() {
+  xy_stream=$1
+  cluster_service_version_yaml=config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
+  
+  if [[ $xy_stream = "x" ]]; then 
+    stream_var=2
+  else
+    exit 
+  fi 
+  
+  # Locate the OCP version from cluster_service_version.yaml
+  version=$(grep -o -E 'OCP \b([0-9]\.[0-9]{2})\b' $cluster_service_version_yaml | 
+    sed -E 's/OCP ([0-9]\.[0-9]{2})/\1/')
+
+  # Increment the version 
+  updated_version=$(echo $version | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  # Find the last version in the dockerfiles 
+  last_version=$(sed -n 's/.*\([0-9]\.[0-9][0-9]\).*/\1/p' build/Dockerfile)
+
+  echo "OCP version $version detected"
+
+  echo "Updating OCP version to $updated_version"
+  sed -i "s/$version/$updated_version/g" $cluster_service_version_yaml
+
+  echo "Updating Dockerfile version to $version"
+  sed -i "s/$last_version/$version/g" bundle.Dockerfile 
+
+  echo "Updating Dockerfile version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile
+
+  echo "Updating Dockerfile.base version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.base
+
+  echo "Updating Dockerfile.ci version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.ci
+
+  echo "Updating Dockerfile.wmco version to $version"
+  sed -i "s/$last_version/$version/g" build/Dockerfile.wmco
+}
+
+if [[ -z $1 ]]; then
+echo "usage:"
+echo "on a release-4.x branch, run ./pre-release (x, y or z)"
+echo "to update a z stream, use ./pre-release.sh z"
+echo "for other streams, use the corresponding x y or z argument."
+fi 
+
+update_WMCO_version $STREAM
```

Comment:
I get an error when I run the script as is
```
$ ./hack/pre-release.sh x
./hack/pre-release.sh: line 107: STREAM: unbound variable
```

Reply:
I get the SkipRange here
https://github.com/openshift/windows-machine-config-operator/blob/195b873a7ebeb69670adbb4290cb7b0e914ad4fd/hack/pre-release.sh#L32 

and update it here 
https://github.com/openshift/windows-machine-config-operator/blob/195b873a7ebeb69670adbb4290cb7b0e914ad4fd/hack/pre-release.sh#L45 
Is there somewhere else I'm missing, or should I add a comment to make it more clear ? 

I was thinking of keeping it to that one version because ideally the versions wouldn't be mixed up like that, and it would be difficult to write regexes for each individual occurrence of each version 
If master has to have that mix of versions though, I could figure it out!

Reply:
When I run `./hack/pre-release.sh x` (after fixing the unbounded variable), the skip version is not being edited. I suspect this is because you are making the changes directly to the bundle CSV in `update_WMCO_version` - when you run `make bundle` at the end of the script it will overwrite such changes because they are not being done to the base CSV

Reply:
Also shouldn't the lower bound of the skipRange also be updated?

Reply:
https://github.com/openshift/windows-machine-config-operator/blob/195b873a7ebeb69670adbb4290cb7b0e914ad4fd/hack/pre-release.sh#L42 
this line updates the upper bound to the new version

and this line https://github.com/openshift/windows-machine-config-operator/blob/195b873a7ebeb69670adbb4290cb7b0e914ad4fd/hack/pre-release.sh#L45 
updates the lower bound to the current version 

I can add a comment on this to make it make more sense

Reply:
Adding here what we talked about in Slack DMs:

bundle csv = bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
base csv = config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml

You can't directly edit the bundle csv. Changes need to be made in the base csv and then they will be applied to the bundle csv with the make bundle command. It's similar to CI release repo - the jobs need to be edited by editing the config and then you run make jobs to get the job file added